zerver/tests/test_subs.py

```diff
@@ -2907,7 +2907,43 @@ def test_remove_invalid_user(self) -> None:
                 "principals": orjson.dumps([99]).decode(),
             },
         )
-        self.assert_json_error(result, "No such user", status_code=400)
+        self.assert_json_error(result, "No such user with given id", status_code=400)
+
+    def test_user_unsubscribe_theirself(self) -> None:
+        """
+        User Trying to unsubscribe theirself from the stream.
+        where principals has the id of the acting_user performing the unsubscribe action.
+        """
+        admin = self.example_user("iago")
+        self.login_user(admin)
+        self.assertTrue(admin.is_realm_admin)
+
+        stream_name = "hümbüǵ"
+        self.make_stream(stream_name)
+        self.subscribe(admin, stream_name)
+
+        # unsubscring when subscribed.
```

Comment:
This is a typo; you might want to install a spellchecker in your code editor to avoid this sort of thing.

(I'm a bit surprised `tools/run-codespell` didn't flag this).